### PR TITLE
Use analyzer input and output

### LINKED ISSUE #3: Add --input option
This will allow users to point to a directory where application source code or binary could be found.

### LINKED ISSUE #4: Add --output option
This will allow users to specify an output directory for the static report as well as the raw YAML output from analyzers

### LINKED ISSUE #5: Add --source option
This will allow users to select from one or more sources from packaged rulesets. This needs to have validation such that only the discovered targets should be allowed.

### LINKED ISSUE #6: Add --target option
This will allow users to select from one or more targets from packaged rulesets. This needs to have validation such that only the discovered targets should be allowed.

### LINKED ISSUE #10: Add --rules option
We need this so that users can pass custom rules along with pre-packaged sources / targets

### PR DESCRIPTION
**Pull request by eemcmullan**
Closes #3 
Closes #4 
Closes #5 
Closes #6 
Closes #10

### CODE REVIEW

File: cmd/analyze.go [hunk 1]
```diff
@@ -175,8 +191,55 @@ func listOptionsFromLabels(sl []string, label string) {
...
+
+	f, err := os.Create(trimmedOutputDir + "/output.yaml")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	return nil
+}
+
+func (a *analyzeCommand) Run(ctx context.Context) error {
```
Comment (pranavgaikwad) [inline cmd/analyze.go:212]: Run() will always execute even when I am using `--list-sources` right? we probably dont want that?

File: cmd/settings.go [hunk 2]
```diff
@@ -10,6 +10,10 @@ const (
 	RulesetPath            = "/opt/rulesets"
 	OpenRewriteRecipesPath = "/opt/openrewrite"
 	InputPath              = "/opt/input"
+	OutputPath             = "/opt/output"
+	OutputFilePath         = "/opt/output/output.yaml"
+	SourceRepoPath         = "/opt/example"
```
Comment (pranavgaikwad) [inline cmd/settings.go:15]: why not re-use InputPath for all inputs? you will only run one command at a time, so shouldn't be an issue

File: settings.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
+[
```
Comment (pranavgaikwad) [inline settings.json:1]: should create provider settings using only the API. this quickly becomes stale when there's a change in the config

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]

File: cmd/analyze.go (hunk 1 above)
Comment (eemcmullan) [reply cmd/analyze.go:212]: Good point. Will update

File: settings.json (hunk 3 above)
Comment (eemcmullan) [reply settings.json:1]: Makes sense. Will update

File: cmd/analyze.go [hunk 4]
```diff
@@ -175,8 +192,94 @@ func listOptionsFromLabels(sl []string, label string) {
...
 		fmt.Println(tech)
 	}
 }
+
+func (a *analyzeCommand) createOutputFile() (string, error) {
+	// if trailing '/' in given output dir, remove it
+	trimmedOutput := strings.TrimRight(a.output, "/")
+	trimmedOutput = strings.TrimRight(a.output, "\\")
+
+	fp := filepath.Join(trimmedOutput, "outout.yaml")
```
Comment (pranavgaikwad) [inline cmd/analyze.go:205]:
```suggestion
	fp := filepath.Join(trimmedOutput, "output.yaml")
```

File: cmd/analyze.go [hunk 5]
```diff
@@ -175,8 +192,94 @@ func listOptionsFromLabels(sl []string, label string) {
...
 	}
-
 	for _, tech := range newSl {
 		fmt.Println(tech)
 	}
 }
+
+func (a *analyzeCommand) createOutputFile() (string, error) {
+	// if trailing '/' in given output dir, remove it
+	trimmedOutput := strings.TrimRight(a.output, "/")
```
Comment (pranavgaikwad) [inline cmd/analyze.go:202]: this is redundant, Join() takes care of everything

File: cmd/analyze.go [hunk 6]
```diff
@@ -26,86 +31,99 @@ var (
...
+				return err
+			}
+			return nil
+		},
+	}
+	analyzeCommand.Flags().BoolVar(&analyzeCmd.listSources, "list-sources", false, "List rules for available migration sources")
+	analyzeCommand.Flags().BoolVar(&analyzeCmd.listTargets, "list-targets", false, "List rules for available migration targets")
+	analyzeCommand.Flags().StringArrayVarP(&analyzeCmd.sources, "source", "s", []string{}, "Source technology to consider for analysis")
+	analyzeCommand.Flags().StringArrayVarP(&analyzeCmd.targets, "target", "t", []string{}, "Target technology to consider for analysis")
+	analyzeCommand.Flags().StringVar(&analyzeCmd.rules, "rules", "", "Rules for analysis")
```
Comment (pranavgaikwad) [inline cmd/analyze.go:72]: I think this should be an StringArray() variable, but fine for now

**Review by pranavgaikwad — APPROVED** [review]
think `ouput.yaml` typo needs fixing, other than that, feel free to address in follow up. Great PR! Thank you!

File: cmd/analyze.go (hunk 6 above)
Comment (eemcmullan) [reply cmd/analyze.go:72]: created issue to address shortly

**Pull request merged**